services/fncsgossbridge/service/fncs_goss_bridge.py

```diff
@@ -906,26 +946,42 @@ def _byteify(data, ignore_dicts = False):
     return data
  
  
-def _keep_alive():
-    while stop_simulation == False:
+def _keep_alive(is_realtime):
+    simulation_ran = False
+    while goss_listener_instance.stop_simulation == False:
         time.sleep(0.1)
+        if goss_listener_instance.start_simulation == True and simulation_ran == False:
+            goss_listener_instance.run_simulation(is_realtime)
+            simulation_ran = True
    
         
-def _main(simulation_id, simulation_broker_location='tcp://localhost:5570', measurement_map_dir=''):
+def _main(simulation_id, simulation_broker_location='tcp://localhost:5570', measurement_map_dir='', is_realtime=True, sim_duration=86400):
  
     measurement_map_file=str(measurement_map_dir)+"model_dict.json"
-    _register_with_goss(simulation_id,'system','manager',goss_server='127.0.0.1',stomp_port='61613')
+    _register_with_goss(simulation_id,'system','manager',goss_server='127.0.0.1',stomp_port='61613', sim_duration)
     _register_with_fncs_broker(simulation_broker_location)
     _create_cim_object_map(measurement_map_file)
-    _keep_alive()
+    _keep_alive(is_realtime)
+    
+def _get_opts():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("simulation_id", help="The simulation id to use for responses on the message bus.")
+    parser.add_argument("broker_location", help="The location of the FNCS broker.")
+    parser.add_argument("simulation_directory", help="The simulation files directory.")
```

Comment:
is this a breaking change simulation_directory I don't remember seeing that in anything before.

Reply:
no it was always in the bridge. it gives the bridge the directory where to find model_dict.json